### DISCUSSION TITLE
Coupled equations formulation for non-linear problem involving interface curvature

### DISCUSSION THREAD

**Discussion by plpiv** [usnistgov/fipy#908] [Q&A] [answered]
Hello,
I'm new to FiPy, and so far very happy about this solver !

I'm modelling the flow of a thin viscous film under lubrification approximation. The local thickness $h$ of the film obeys the following non-linear equation:
$\frac{\partial h}{\partial t}=\nabla \cdot (\frac{-\gamma h^3}{3\eta} \nabla\kappa) $
where $\kappa$ is the local curvature.

When considering the simplified form of the curvature ($\kappa = \nabla^2 h$) for small modulations, the equation is easy to put in FiPy canonical form (1D-case):
`eq= ( TransientTerm() == DiffusionTerm(coeff= (-gamma*h**3/(3*eta),1)) )`

I want to use the full expression of the curvature:
$\kappa = \nabla \cdot(\frac{1}{\sqrt{1+(\lVert \nabla h \rVert)^2}}\nabla h)$ **[eq.2]**

How to formulate this in a FiPy tractable equation (in 1D) ?
I thought of using a system of two coupled equations (on $h$ and $\kappa$),
with the first equation being simply:
`eq= ( TransientTerm(coeff=1., var=h) == DiffusionTerm(coeff= -gamma*h**3/(3*eta),var=kappa )`
but the second equation on $\kappa$ (eq. 2) can't be handled by FiPy I guess?
What would be a correct and efficient formulation ?

Thanks!

**Comment by guyer** [comment, marked as answer]
I would think you could write
```python
h_eq = TransientTerm(coeff=1., var=h) == DiffusionTerm(coeff= -gamma*h**3/(3*eta), var=kappa)
kappa_eq = ImplicitSourceTerm(coeff=1., var=kappa) == DiffusionTerm(coeff=1/numerix.sqrt(1 + (h.faceGrad.mag)**2), var=h)
eq = h_eq & kappa_eq
```
See, e.g, the [coupled Cahn-Hilliard example](https://www.ctcms.nist.gov/fipy/examples/cahnHilliard/generated/examples.cahnHilliard.mesh2DCoupled.html).

**Reply by plpiv** [reply]
Thanks! That works well !
